src/link/mod.rs

```diff
@@ -119,47 +127,60 @@ impl Link {
         self.thread_handles.push(recv_thread);
     }
 
-    pub fn stop(&mut self) {
+    pub fn stop(&mut self) -> Result<(), AetherError>{
         // Set the stop flag
-        let mut flag_lock = self.stop_flag.lock().expect("Unable to lock stop flag");
-        *flag_lock = true;
+        match self.stop_flag.lock() {
+            Ok(ref mut flag_lock) => {
```

Comment:
`flag_lock` doesn't need to be a reference. We should change this to just `mut flag_lock`
Otherwise, `drop(flag_lock)` would do nothing.